### PR TITLE
Separates dev and production npm modules

### PR DESCRIPTION
**Pull request by anselmbradford**
Separates dev and production npm modules.

## Changes

- Separates out production and development npm modules. 
- Allows passing of arguments to both frontend.sh and backend.sh scripts to specify development, test, or production environments.

## Testing

- See screenshots for commands.

## Review

- @kave 
- @KimberlyMunoz 
- @sebworks 
- @jimmynotjim 

## Screenshots

screen shot 2015-12-16 at 2 16 23 pm

screen shot 2015-12-16 at 2 16 13 pm

screen shot 2015-12-16 at 2 16 35 pm

screen shot 2015-12-16 at 2 16 17 pm

screen shot 2015-12-16 at 2 16 28 pm

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,51 +7,54 @@
...
-    "gulp-load-plugins": "^1.0.0-rc.1",
-    "gulp-mocha": "^2.1.3",
+    "gulp-load-plugins": "^1.1.0",
+    "gulp-imagemin": "^2.3.0",
     "gulp-mq-remove": "0.0.2",
     "gulp-notify": "^2.2.0",
-    "gulp-plumber": "^1.0.1",
-    "gulp-rename": "^1.2.2",
-    "gulp-replace": "^0.5.3",
     "gulp-sitespeedio": "^0.0.8",
```
Comment (kave) [inline package.json]: is this still necessary? thought it was only used for `gulp test:pref`

Comment (anselmbradford) [inline package.json]: Sadly so, something is still calling for it :( I moved the modules module-by-module and in the current setup this is how it fleshing out. I'd say we get this PR in and then in the remaining modules in the `dependencies` object in `package.json`, flush out why some of the test modules are still required in the production environment.

Comment (anselmbradford) [inline package.json]: This is the error when running `setup.sh production` and having `gulp-sitespeedio` outside of `dependencies`:

```
Building project...
module.js:329
    throw err;
    ^

Error: Cannot find module 'gulp-sitespeedio'
    at Function.Module._resolveFilename (module.js:327:15)
    at Function.Module._load (module.js:278:25)
    at Module.require (module.js:355:17)
    at require (internal/module.js:13:17)
    at Object.<anonymous> (/Users/bradforda/Projects/cfgov-refresh/gulp/tasks/test.js:5:19)
    at Module._compile (module.js:399:26)
    at Object.Module._extensions..js (module.js:406:10)
    at Module.load (module.js:345:32)
    at Function.Module._load (module.js:302:12)
    at Module.require (module.js:355:17)
    at require (internal/module.js:13:17)
```

**Comment by kave** [comment]
Heads up travis failure is due to conflicting migrations, a fix will be in soon when https://github.com/cfpb/cfgov-refresh/pull/1227 merges

File: package.json (hunk 1 above)
Comment (kave) [inline package.json]: alright sounds good

**Comment by kave** [comment]
:+1:

**Pull request merged**